htdocs/margin/tabs/thirdpartyMargins.php

```diff
@@ -226,8 +229,16 @@
     			print '<td class="right">'.$invoicestatic->LibStatut($objp->paye, $objp->statut, 5).'</td>';
     			print "</tr>\n";
     			$i++;
-    			$cumul_vente += $objp->selling_price;
-    			$cumul_achat += ($objp->type == 2 ? -1 : 1) * $objp->buying_price;
+				
+				if ($objp->buying_price != 0 ) { //Ne selectionne que les lignes où le prix de revient est renseigné
```

Comment:
It would be better to report a warning on screen to say some invoice has no buying_price than excluding data for total of sales. Because this will lead to not consistent data compared to other screens.

Reply:
Ok, if you want.